compiler-plugin/src/main/java/io/ballerina/stdlib/ai/plugin/GenerateMethodModificationTask.java

```diff
@@ -124,14 +127,57 @@ public void modify(SourceModifierContext modifierContext) {
         }
     }
 
-    private void analyzeDocument(Module module, DocumentId documentId, SemanticModel semanticModel) {
+    private Optional<ModuleSymbol> getAiModuleSymbolFromProjectDocuments(Module module, SemanticModel semanticModel,
+                                          Collection<DocumentId> documentIds, Collection<DocumentId> testDocumentIds) {
+        Optional<ModuleSymbol> aiModuleSymbol = getAiModuleSymbolFromDocuments(module, semanticModel, documentIds);
+        if (aiModuleSymbol.isPresent()) {
+            return aiModuleSymbol;
+        }
+
+        return getAiModuleSymbolFromDocuments(module, semanticModel, testDocumentIds);
+    }
+
+    private Optional<ModuleSymbol> getAiModuleSymbolFromDocuments(Module module, SemanticModel semanticModel,
+                                        Collection<DocumentId> documentIds) {
+        Optional<ModuleSymbol> aiModuleSymbol;
+        for (DocumentId documentId : documentIds) {
+            Document document = module.document(documentId);
+            Node rootNode = document.syntaxTree().rootNode();
+            aiModuleSymbol = checkAndReturnAiModelProviderSymbol(
+                    semanticModel.visibleSymbols(document, rootNode.lineRange().startLine()));
+
+            if (aiModuleSymbol.isPresent()) {
+                return aiModuleSymbol;
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<ModuleSymbol> checkAndReturnAiModelProviderSymbol(List<Symbol> symbols) {
+        for (Symbol symbol : symbols) {
+            if (!(symbol instanceof ModuleSymbol moduleSymbol)) {
+                continue;
+            }
+
+            ModuleID id = moduleSymbol.id();
+            if (BALLERINA_ORG_NAME.equals(id.orgName())
+                    && AI_MODULE_NAME.equals(id.moduleName())
+                    && id.version().startsWith(AI_MODULE_MAJOR_VERSION)) {
+                return Optional.of(moduleSymbol);
+            }
+        }
+        return Optional.empty();
+    }
```

Comment:
Can't you just do https://github.com/ballerina-platform/module-ballerina-ai.np/blob/v0.3.0/compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/Validator.java#L79?